runelite-client/src/main/java/net/runelite/client/plugins/worldmap/WorldMapPlugin.java

```diff
@@ -185,7 +185,7 @@ public void onStatChanged(StatChanged statChanged)
 				if (newAgilityLevel != agilityLevel)
 				{
 					agilityLevel = newAgilityLevel;
-					updateAgilityIcons();
+					clientThread.invokeLater(this::updateAgilityIcons);
```

Comment:
Rather than requiring callsites be aware of the client thread requirement of this method, move the relevant code inside a `clientThread.invokeLater(() -> {})` block.